javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -82,17 +82,66 @@ predicate alwaysThrows(Function f) {
   )
 }
 
-from DataFlow::CallNode call
-where
+predicate callToVoidFunction(DataFlow::CallNode call, Function func) {
   not call.isIndefinite(_) and 
-  forex(Function f | f = call.getACallee() |
+  func = call.getACallee() and
+  forall(Function f | f = call.getACallee() |
     returnsVoid(f) and not isStub(f) and not alwaysThrows(f)
+  )
+}
+
+predicate hasNonVoidCallbackMethod(string name) {
+  name = "every" or
+  name = "filter" or
+  name = "find" or
+  name = "findIndex" or
+  name = "flatMap" or
+  name = "map" or
+  name = "reduce" or
+  name = "reduceRight" or
+  name = "some" or
+  name = "sort"
+}
+
+DataFlow::SourceNode array(DataFlow::TypeTracker t) {
+  t.start() and result instanceof DataFlow::ArrayCreationNode
+  or
+  exists (DataFlow::TypeTracker t2 |
+    result = array(t2).track(t2, t)
+  )
+}
+
+DataFlow::SourceNode array() { result = array(DataFlow::TypeTracker::end()) }
+
+predicate voidArrayCallback(DataFlow::MethodCallNode call, Function func) {
+  hasNonVoidCallbackMethod(call.getMethodName()) and
+  func = call.getAnArgument().getALocalSource().asExpr() and
+  1 = count(DataFlow::Node arg | arg = call.getAnArgument() and arg.getALocalSource().asExpr() instanceof Function) and
```

Comment:
Is this still necessary with the `array()` check? As far as I can tell, none of the methods we flag here could reasonably take two functions as arguments.

Reply:
Yeah, you're right.  
I misunderstood [some instances I found where multiple functions were used](https://lgtm.com/query/4020504995801656884/).  
In those there is only 1 callback function, and the second function argument is the initial value for the `reduce` operation. 

The check should be replaced with a check that ensures that the callback is the first function argument given.

Reply:
Do you still want to address this?

Reply:
Yes, I somehow overlooked this when looking back at this issue.